src/components/Home/CarouselContainer.tsx

```diff
@@ -0,0 +1,103 @@
+import { useState, useEffect } from "react";
+
+import { styled } from "styled-components";
+
+import { IcCheckLeft, IcCheckRight } from "../../assets";
+import IcCsOne from "../../assets/img/img_carousel_1.png";
+import IcCsTwo from "../../assets/img/img_carousel_2.png";
+import IcCsThree from "../../assets/img/img_carousel_3.png";
+import IcCsFour from "../../assets/img/img_carousel_4.png";
+import { CarouselImgContainerProps } from "../../types/Home/homeTypes";
+
+const images = [IcCsOne, IcCsTwo, IcCsThree, IcCsFour];
+const totalImages = [...images, ...images, ...images];
+
+function CarouselContainer() {
+  const [offset, setOffset] = useState(-images.length * 573);
+  const [transition, setTransition] = useState(true);
+  const imageWidth = 573;
+
+  const handleLeftClick = () => {
+    setOffset((prevOffset) => prevOffset + imageWidth);
+    setTransition(true);
+  };
+
+  const handleRightClick = () => {
+    setOffset((prevOffset) => prevOffset - imageWidth);
+    setTransition(true);
+  };
+
+  const handleTransitionEnd = () => {
+    if (offset >= 0) {
+      setOffset(-imageWidth * images.length);
+      setTransition(false);
+    } else if (offset <= -imageWidth * images.length * 2) {
+      setOffset(-imageWidth * images.length);
+      setTransition(false);
+    }
+  };
+
+  useEffect(() => {
+    if (!transition) {
+      const id = setTimeout(() => {
+        setTransition(true);
+      }, 50);
+      return () => clearTimeout(id);
+    }
+  }, [transition]);
+
+  return (
+    <CarouselContainerWrapper>
+      <CarouselImgContainer offset={offset} transition={transition} onTransitionEnd={handleTransitionEnd}>
+        {totalImages.map((src, index) => (
+          <CarouselImage src={src} alt={`Slide ${index}`} key={index} />
+        ))}
+      </CarouselImgContainer>
+      <LeftButton onClick={handleLeftClick}>
+        <IcLeftBtn />
+      </LeftButton>
+      <RightButton onClick={handleRightClick}>
+        <IcRightBtn />
+      </RightButton>
+    </CarouselContainerWrapper>
+  );
+}
+
+export default CarouselContainer;
+
+const CarouselContainerWrapper = styled.section`
+  display: flex;
+  overflow: hidden;
+  position: relative;
+  width: 100vw;
+  height: 30.3rem;
+  margin-top: 4rem;
+`;
+
+const CarouselImgContainer = styled.section<CarouselImgContainerProps>`
+  display: flex;
+  transform: translateX(${(props) => props.offset}px);
+  transition: ${(props) => (props.transition ? "transform 0.5s ease-in-out" : "none")};
```

Comment:
사소하지만, 구조 분해 할당과 단락 평가를 이용하면 코드를 더 간소화시킬 수 있어요!!
추가적으로 우리 px 안 쓰기로 했으니까 rem 단위로 맞춰줘도 좋을 것 같아요!!
```suggestion
  transform: translateX(${({ offset }) => offset / 10}rem);
  transition: ${({transition}) => (transition && "transform 0.5s ease-in-out")};
```

사실 여기서 10으로 나눠서 rem 값으로 맞춰주는 것보다는, CarouselImgContainer로 넘어오는 offset 값을 애초에 10으로 나눈 값으로 넘겨주는 게 역할상 더 맞습니다!! 스타일 컴포넌트는 단순 "스타일"을 주는 역할이지 "연산"을 하는 역할은 아니기 때문이에요!!

Reply:
> 사소하지만, 구조 분해 할당과 단락 평가를 이용하면 코드를 더 간소화시킬 수 있어요!! 추가적으로 우리 px 안 쓰기로 했으니까 rem 단위로 맞춰줘도 좋을 것 같아요!!
> 
> 사실 여기서 10으로 나눠서 rem 값으로 맞춰주는 것보다는, CarouselImgContainer로 넘어오는 offset 값을 애초에 10으로 나눈 값으로 넘겨주는 게 역할상 더 맞습니다!! 스타일 컴포넌트는 단순 "스타일"을 주는 역할이지 "연산"을 하는 역할은 아니기 때문이에요!!

미쳐 생각하지 못했던 부분인데 감사합니다! 적극 반영 하겠습니다.